app/api/csv/csvLoader.ts

```diff
@@ -56,7 +56,16 @@ export class CSVLoader extends EventEmitter {
       (await settings.get()).languages
     ).map((l: LanguageSchema) => l.key);
     const { newNameGeneration = false } = await settings.get();
-    await arrangeThesauri(file, template, availableLanguages, this);
+    try {
+      await arrangeThesauri(file, template, availableLanguages, this.stopOnError);
+    } catch (e) {
+      if (e instanceof ArrangeThesauriError) {
+        const _e: ArrangeThesauriError = e;
+        this.emit('loadError', _e.source, toSafeName(_e.row), _e.index);
+      } else {
+        throw e;
+      }
+    }
```

Comment:
@LaszloKecskes this whole try/catch block i believe is not tested, only the "no error" portion is. Also this is adding an eslint error for the current function having to many statements, maybe move the block into its own method?

Reply:
Removed during pairing.